tool/tctl/common/terraform_command.go

```diff
@@ -0,0 +1,380 @@
+// - exports certificates and Terraform configuration in environment variables
+// envOutput and userOutput parameters are respectively stdout and stderr,
+// except during tests where we want to catch the command output.
+func (c *TerraformCommand) RunEnvCommand(ctx context.Context, client *authclient.Client, envOutput, userOutput io.Writer) error {
+	// If we're not actively debugging, suppress any kind of logging from other teleport components
+	if !c.cfg.Debug {
+		utils.InitLogger(utils.LoggingForCLI, slog.LevelError)
+	}
+	c.envOutput = envOutput
+	c.userOutput = userOutput
+
+	// Validate that the bot expires
+	if c.botTTL == 0 {
+		return trace.BadParameter("--bot-ttl must be greater than zero")
+	}
+
+	addrs := c.cfg.AuthServerAddresses()
+	if len(addrs) == 0 {
+		return trace.BadParameter("no auth server addresses found")
+	}
+	addr := addrs[0]
+
+	// Prompt for admin action MFA if required, allowing reuse for UpsertRole, UpsertToken and CreateBot.
+	c.showProgress("Detecting if MFA is required")
+	mfaResponse, err := mfa.PerformAdminActionMFACeremony(ctx, client.PerformMFACeremony, true /*allowReuse*/)
+	if err == nil {
+		ctx = mfa.ContextWithMFAResponse(ctx, mfaResponse)
+	} else if !errors.Is(err, &mfa.ErrMFANotRequired) && !errors.Is(err, &mfa.ErrMFANotSupported) {
+		return trace.Wrap(err)
+	}
+
+	// Upsert Terraform role
+	roleName, err := c.createRoleIfNeeded(ctx, client)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Create temporary bot and token
+	tokenName, err := c.createTransientBotAndToken(ctx, client, roleName)
+	if err != nil {
+		return trace.Wrap(err, "bootstrapping bot")
+	}
+
+	// Now run tbot
+	c.showProgress("Using the temporary bot to obtain certificates 🤖")
+	id, err := c.useBotToObtainIdentity(ctx, addr, tokenName, client)
+	if err != nil {
+		return trace.Wrap(err, "obtaining identity")
+	}
+
+	envVars, err := identityToTerraformEnvVars(addr.String(), id)
+	if err != nil {
+		return trace.Wrap(err, "exporting identity into environment variables")
+	}
+
+	// Export environment variables
+	c.showProgress("Certificates obtained, you can now use Terraform in this terminal 🚀")
+	for env, value := range envVars {
+		_, _ = fmt.Fprintf(c.envOutput, "export %s=%q\n", env, value)
+	}
+	fmt.Println("# You must invoke this command in an eval: eval $(tctl terraform-helper)")
+	return nil
+}
+
+// createTransientBotAndToken creates a Bot resource and a secret Token.
+// The token is single use (secret tokens are consumed on MachineID join)
+// and the bot expires after the given TTL.
+func (c *TerraformCommand) createTransientBotAndToken(ctx context.Context, client *authclient.Client, roleName string) (string, error) {
+	// Create token and bot name
+	suffix, err := utils.CryptoRandomHex(4)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	botName := c.resourcePrefix + suffix
+	c.showProgress(fmt.Sprintf("Creating temporary bot %q and its token", botName))
+
+	roles := []string{roleName}
```

Comment:
Can we please define the `roles` var right next to where it is used? Or drop the initialization entirely and just rule `[]string{roleName}` directly?